provider/openstack/cinder.go

```diff
@@ -665,3 +660,29 @@ func (ga *openstackStorageAdapter) GetVolume(volumeId string) (*cinder.Volume, e
 func (ga *openstackStorageAdapter) SetVolumeMetadata(volumeId string, metadata map[string]string) (map[string]string, error) {
 	return ga.cinderClient.SetVolumeMetadata(volumeId, metadata)
 }
+
+// DeleteVolume is part of the OpenstackStorage interface.
+func (ga *openstackStorageAdapter) DeleteVolume(volumeId string) error {
+	if err := ga.cinderClient.DeleteVolume(volumeId); err != nil {
+		if gooseerrors.IsNotFound(err) {
+			return errors.NotFoundf("volume %q", volumeId)
```

Comment:
I'm not quite sure why we are translating NotFound into NotFound. Are they different objects?
Aren't they all derived from juju/errors ?

Are we attaching more information here?

Reply:
> Aren't they all derived from juju/errors ?

Nope. goose/errors doesn't use juju/errors.

I was just going to use gooseerrors.IsNotFound up the stack, but goose/errors.gooseError happens to implement juju/errors.causer, which leads to the gooseError bit not being the "cause" when the error is traced/annotated.